src/net/stackerdb/mod.rs

```diff
@@ -0,0 +1,133 @@
+/// Stacks-powered applications are able to leverage the Stacks peer-to-peer node network to host
+/// and dissiminate their data without incuring the cost and performance penalties of bundling it
+/// within a transaction.
+///
+/// ## Data Model
+///
+/// Data within a StackerDB is eventually-consistent.  In the absence of writes and network
+/// partitions, all replicas will receive the latest data in a finite number of protocol rounds,
+/// with high probability.  Given that network partitions in the peer-to-peer network are assumed
+/// to be temporary, we assume that all StackerDB instances will receive the latest state in finite time.
+/// Beyond this, it makes no guarantees about how quickly a write will materialize on a given replica.
+///
+/// The StackerDB schema is chunk-oriented.  Each StackerDB contains a fixed number of fixed-size
+/// chunks.  A `write` to a StackerDB is the act of replacing one chunk's data with new data, and a
+/// `read` on a StackerDB is the act of loading one chunk from the node's local replica.  Reading
+/// and writing a single chunk on one node is atomic.  StackerDB replication proceeds in a
+/// store-and-forward manner -- newly-discovered chunks are stored to the node's local replica and
+/// broadcast to a subset of neighbors who also replicate the given StackerDB.
+///
+/// Each chunk has an associated Lamport clock, and an associated public key hash used to
+/// authenticate writes.  The Lamport clock is used to identify the latest version of a chunk --
+/// a node will replace an existing but stale copy of a chunk with a newly-downloaded chunk if its
+/// Lamport clock has a strictly higher value.  The chunk's metadata -- its ID, Lamport clock, and
+/// data hash -- must be signed by the chunk's public key hash's associated private key in order to
+/// be stored.  The chunks themselves are ordered byte sequences with no mandatory internal
+/// structure.
+///
+/// StackerDB state is ephemeral.  Its contents are dropped at the start of every reward cycle.
+/// Endpoints must re-replicate data to the StackerDB if they wish to keep it online.  In doing so,
+/// the set of StackerDBs is self-administrating -- a node will only store state for active
+/// StackerDBs.
+///
+/// ## Control Plane
+///
+/// The smart contract to which a StackerDB is bound controls how many chunks the DB has, who can
+/// write to which chunks (identified by public key hash), how big a chunk is, and how often a
+/// chunk can be written to (in wall-clock time).  This smart contract is queried once per reward cycle
+/// in order to configure the database.  The act of configuring the re-configuring the database
+/// is also the act of dropping and reinstantiating it.
+///
+/// Applications that employ StackerDBs would deploy one or more smart contracts that list out
+/// which users can store data to the StackerDB replica, and how much space they get.
+///
+/// ## Replication Protocol
```

Comment:
This section doesn't mention `stackerdb_push` messages. What are those?

Reply:
It's not implemented in this PR.